vue/mixins/logic.js

```diff
@@ -12,71 +346,210 @@ export const logicMixin = {
             // in case the config is not explicitly defined "computes" the
             // best possible decision on if the instance should be configured
             const isNewInstance = Boolean(!this.ripeData && !global.ripe);
-            const config = this.config === null ? isNewInstance : this.config;
+            this.configData = this.config === null ? isNewInstance : this.config;
 
             // in case there's no internal RIPE instance already
             // available then created a new one with default config
             if (!this.ripeData) {
                 this.ripeData = global.ripe || new Ripe();
             }
 
+            this.onPreConfig = this.ripeData.bind("pre_config", async (brand, model, options) => {
+                await this._copyRipeData();
+            });
+
+            this.onPostConfig = this.ripeData.bind("post_config", async (loadedConfig, options) => {
+                await this._copyRipeData();
+            });
+
+            this.onParts = this.ripeData.bind("parts", async parts => {
+                if (this.equalParts(parts, this.partsData)) return;
+                if (this.structure) {

```

Comment:
Can't this be `structureData`?

Reply:
I believe it can, although it will result in the same thing, since if there is a structure, the structureData will be not null after the first _copyRipeData